### ISSUE TITLE
Don't use window constructors in instance checks

### ISSUE THREAD

**Issue by Joozty** [pelotoncycle/resize-observer#98]
Constructors are not shared between window instances. So for example if you have an HTML element in an iframe this check will fail:

```
iframeDiv instanceof window.Element
```

Read more [here](https://stackoverflow.com/a/6473338/5594539) 

I suggest using a different strategy to check if the[ given parameter is an Element](https://github.com/pelotoncycle/resize-observer/blob/master/src/ResizeObserver.ts#L89). For example, using [tagName](https://developer.mozilla.org/en-US/docs/Web/API/Element/tagName)

MWE:

```
import { ResizeObserver } from "resize-observer";

document.getElementById("app").innerHTML = `
<iframe></iframe>
`;

const resizeObserver = new ResizeObserver((entries) => {
  for (let entry of entries) {
    console.log(entry);
  }
});

const iframe = document.querySelector("iframe");

resizeObserver.observe(iframe.contentDocument.body);

iframe.contentDocument.body.style.width = "300px";
```

Codesanbox [here](https://codesandbox.io/s/busy-gagarin-3nlyq?file=/src/index.js:0-394)

**Comment by devrelm** [comment]
Great catch, and an interesting problem.

I would like to avoid duck-typing, so I looked up a way to get the window that an element is in using [`defaultView`](https://developer.mozilla.org/en-US/docs/Web/API/Document/defaultView):

```js
target instanceof target.ownerDocument.defaultView.Element
```

I'll have a PR open shortly.

**Comment by Joozty** [comment]
That won't solve it. I am afraid there is no better solution than `tagName`. You could store all window object to some array and then check instanceof but it is much more complicated.

See:

```
const iframe = document.createElement("iframe");
document.body.append(iframe);
const divElement = iframe.contentDocument.createElement("div");

document.body.append(divElement);

console.log(divElement instanceof divElement.ownerDocument.defaultView.Element);
console.log(divElement.ownerDocument)
```

**Comment by Joozty** [comment]
It's not really my use case but there are some applications that work like that. I work on a script that is used on many websites and from the error logs I can see that some websites mix elements between iframes. Obviously, it's not widely used but still the real ResizeObserver works with the example I provided above so I guess the polyfill should be as close as possible to it especially when the fix is quite easy.

**Comment by devrelm** [comment]
Fair enough. I'm out of time to work on this today, but how about instead of checking `tagName` we go with checking `target.nodeType === Node.ELEMENT_NODE`.

**Comment by Joozty** [comment]
that's even better

**Comment by devrelm** [comment]
This has been fixed in #100, published with v1.0.4.
